pkg/crc/preflight/preflight_linux.go

```diff
@@ -227,17 +227,15 @@ func getNetworkChecksForDistro(distro *linux.OsRelease, networkMode network.Mode
 		return append(checks, daemonRunningChecks)
 	}
 
-	switch {
-	default:
+	if !(distroIsLike(distro, linux.Ubuntu) || distroIsLike(distro, linux.Fedora)) {
 		logging.Warnf("distribution-specific preflight checks are not implemented for '%s'", distro.ID)
-		fallthrough
-	case distroIsLike(distro, linux.Ubuntu), distroIsLike(distro, linux.Fedora):
-		checks = append(checks, nmPreflightChecks[:]...)
-		if usesSystemdResolved(distro) {
-			checks = append(checks, systemdResolvedPreflightChecks[:]...)
-		} else {
-			checks = append(checks, dnsmasqPreflightChecks[:]...)
-		}
+	}
+
+	checks = append(checks, nmPreflightChecks[:]...)
+	if checkSystemdResolvedIsRunning() == nil {
```

Comment:
The integration tests are failing because of this logic now before as part of `uessSystemResolved` function we explicit mentioned if `distroIsLike` ubuntu or `fedora>= 33` then we should include this reolved preflight check but now unit test depend on the system where it is executed and `crcSystemResolvedIsRunning` deciding factor.

Reply:
`distro` argument is not needed anymore in this function in fact. You can remove it from here. 

For fixing the tests, I would: 
* Add a bool like `systemdResolved bool` to `getPreflightChecksForDistro` function. 
* Pass `checkSystemdResolvedIsRunning()` in  `getPreflightChecks`
* Add a new field like `systemdResolved bool` in:

```
type checkListForDistro struct {
	distro      *crcos.OsRelease
	networkMode network.Mode
	checks      []Check
}

```
* Update test cases accordingly. 

No worry, if you don't have time to do it. Let me know I can do it.

Reply:
I'll take a shot at it and see what happens! (also, re: docs, the dispatcher script would also need to be modified to be the `systemd-resolve` command, rather than `resolvectl`).

Reply:
@guillaumerose I take it I'm going to have to duplicate each test case that uses `network.DefaultMode`?

Also, `checkSystemdResolvedIsRunning()` returns an `error` rather than a bool, should I do a conversion internal to the calling function, i.e. `if cSRIR() == null: otherVar := true else false`, and pass that to `getPreflightChecksForDistro` or will the `error` be fine?

Just want to be sure of what you consider best practice for this.